assembly/src/ast/procedure/name.rs

```diff
@@ -328,3 +333,47 @@ impl FromStr for ProcedureName {
         Ok(Self(Ident::new_unchecked(Span::unknown(raw))))
     }
 }
+
+impl Serializable for ProcedureName {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.as_str().write_into(target)
+    }
+}
+
+impl Deserializable for ProcedureName {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let str: String = source.read()?;
+        let proc_name =
+            ProcedureName::new_unchecked(Ident::new_unchecked(Span::unknown(Arc::from(str))));
+        Ok(proc_name)
+    }
+}
+
+// TESTS
+// ================================================================================================
+
+/// Tests
+#[cfg(test)]
+mod tests {
+    use vm_core::{
+        debuginfo::Span,
+        utils::{Deserializable, Serializable},
+    };
+
+    use super::ProcedureName;
+    use crate::ast::Ident;
+
+    #[test]
+    fn procedure_name_serialization_rustc_mangled() {
+        // Tests that rustc mangled symbols are serialized and deserialized
+        // see https://doc.rust-lang.org/rustc/symbol-mangling/v0.html#symbol-grammar-summary
```

Comment:
Why do we need to test this explicitly?

Reply:
I put it as a regression test in case an incompatible validation is (re-)introduced in the `ProcedureName` deserialization.

Reply:
I removed this test since this case is covered in the serialization roundtrip test via using these characters for one of the procedure names in `Arbitrary` impl (proptest).